src/kernel/sched/idle_light.c

```diff
@@ -1,29 +1,33 @@
-#include <util/err.h>
+/**
+ * @file
+ * @brief
+ *
+ * @author  Vita Loginova
+ * @date    08.12.2014
+ */
+
 #include <hal/arch.h>
 #include <kernel/lthread/lthread.h>
-#include <kernel/lthread/lthread_priority.h>
 
-static void *idle_run(void *arg) {
+static struct lthread idle;
+
+static int idle_run(struct lthread *self) {
 	arch_idle();
-	sched_wakeup(schedee_get_current());
-	return NULL;
+	lthread_launch(self);
+	return 0;
 }
 
 int idle_thread_create(void) {
-	struct lthread *lt;
 
-	lt = lthread_create(idle_run, NULL);
-	if (0 != err(lt)) {
-		return err(lt);
-	}
+	lthread_init(&idle, idle_run);
 
-	sched_change_priority(&lt->schedee, SCHED_PRIORITY_MIN);
-#if 0 //XXX
+	schedee_priority_set(&idle.schedee, SCHED_PRIORITY_MIN);
+#if 0 //XX
```

Comment:
Does it need for SMP? If yes may be it's better to use macro for this SMP_USE MULTICORE_SUPPORT